File: app/src/main/java/com/dian/demo/repository/local/DatabaseFactory.java
package com.dian.demo.repository.local;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;


/**
 * 样例
 * @Database(entities = {DatabaseEntity.class}, version = 1, exportSchema = false)
 */
public abstract class DatabaseFactory extends RoomDatabase {

    private static DatabaseFactory dataBase;

    public static synchronized DatabaseFactory getInstance(Context mContext) {
        if (dataBase == null) {
            dataBase = Room.databaseBuilder(mContext.getApplicationContext(), DatabaseFactory.class, "db_database_entity")
                    .build();
        }
        return dataBase;
    }

    /**
     *  样例
     *  public abstract DatabaseDao getDatabase();
     */



}

File: app/src/main/java/com/dian/demo/di/repository/local/DatabaseFactory.kt
package com.dian.demo.di.repository.local

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.dian.demo.di.model.ArticleEntity



@Database(entities = [ArticleEntity::class], version = 1, exportSchema = false)
abstract class DatabaseFactory: RoomDatabase() {

    companion object{
        @Volatile
        private var dataBase: DatabaseFactory? = null

        fun getInstance(mContext: Context): DatabaseFactory {
            if (dataBase == null) {
                synchronized(DatabaseFactory::class.java) {
                    if (dataBase == null) {
                        dataBase = Room.databaseBuilder(mContext.applicationContext, DatabaseFactory::class.java, "db_article_entity")
                            .build()
                    }
                }
            }
            return dataBase!!
        }
    }


    abstract fun getArticle(): ArticleDao

}